src/Symfony/Component/Security/Core/Authentication/Provider/AnonymousAuthenticationProvider.php

```diff
@@ -38,7 +38,7 @@ public function __construct($key)
     public function authenticate(TokenInterface $token)
     {
         if (!$this->supports($token)) {
-            return;
+            throw new BadCredentialsException('The token is not supported by this authentication provider.');
```

Comment:
I would use `AuthenticationException`, not `BadCredentialsException`, as it is not a case of bad credentials (it is a case of a bug in the `supports` implementation or in the calling code ignoring `supports`)

Reply:
Thanks, done.